pkg/controllers/machinesync/machine_sync_controller.go

```diff
@@ -254,10 +259,10 @@ func (r *MachineSyncReconciler) Reconcile(ctx context.Context, req reconcile.Req
 		return ctrl.Result{}, nil
 	case authoritativeAPI == "":
 		logger.Info("Machine status.authoritativeAPI is empty, will check again later", "AuthoritativeAPI", mapiMachine.Status.AuthoritativeAPI)
-		return ctrl.Result{}, nil
+		return ctrl.Result{RequeueAfter: requeueUnknownAuthoritativeAPI}, nil
 	default:
 		logger.Info("Machine status.authoritativeAPI has unexpected value", "AuthoritativeAPI", mapiMachine.Status.AuthoritativeAPI)
-		return ctrl.Result{}, nil
+		return ctrl.Result{RequeueAfter: requeueUnexpectedAuthoritativeAPI}, nil
```

Comment:
There's no point requeuing here. If the value changes we'll be reconciled again anyway. If it doesn't change, there's no point in reconciling again because it's invalid.

Reply:
For posterity, this exposed a bug where our watches of infra machines assume CAPI Machine name == Infra machine name. Will be fixed in another follow up PR.